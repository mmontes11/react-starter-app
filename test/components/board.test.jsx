import React from "react";
import { shallow } from "enzyme";
import Board from "components/board";

describe("components/board", () => {
  const squares = ["X", "X", "O", "X", "O", "O", "X", "O", "O"];
  const onClick = jest.fn();
  const wrapper = shallow(<Board squares={squares} onClick={() => onClick()} />);

  it("renders a board", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates clicks", () => {
    wrapper
      .find("Square")
      .first()
      .simulate("click");

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
