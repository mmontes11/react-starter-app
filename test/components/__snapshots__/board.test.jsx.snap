// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/board renders a board 1`] = `
<div>
  <div
    className="board-row"
    key="0"
  >
    <Square
      key="0"
      onClick={[Function]}
      value="X"
    />
    <Square
      key="1"
      onClick={[Function]}
      value="X"
    />
    <Square
      key="2"
      onClick={[Function]}
      value="O"
    />
  </div>
  <div
    className="board-row"
    key="1"
  >
    <Square
      key="3"
      onClick={[Function]}
      value="X"
    />
    <Square
      key="4"
      onClick={[Function]}
      value="O"
    />
    <Square
      key="5"
      onClick={[Function]}
      value="O"
    />
  </div>
  <div
    className="board-row"
    key="2"
  >
    <Square
      key="6"
      onClick={[Function]}
      value="X"
    />
    <Square
      key="7"
      onClick={[Function]}
      value="O"
    />
    <Square
      key="8"
      onClick={[Function]}
      value="O"
    />
  </div>
</div>
`;
